src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +489,106 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
                 return true;
             }
         }
+
+        public static long? TryGetLastRaftIndexForDeleteDatabaseCommandWithoutErrors(ServerStore serverStore, string databaseName, out bool hasExceptions)
+        {
+            var commandType = nameof(DeleteDatabaseCommand);
+            long? index = null;
+            hasExceptions = false;
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+                var guidPrefix = $"{databaseName}/";
+                using (Slice.From(context.Allocator, guidPrefix, out var guidPrefixSlice))
+                {
+                    foreach (var (guidSlice, entryHolder) in table.SeekByPrimaryKeyPrefix(guidPrefixSlice, Slices.Empty, 0))
+                    {
+                        var entry = ReadHistoryLog(context, entryHolder);
+                        var curIndex = long.Parse(entry[nameof(LogHistoryColumn.Index)].ToString());
+                        var guid = entry[nameof(LogHistoryColumn.Guid)].ToString();
+                        var type = entry[nameof(LogHistoryColumn.Type)].ToString();
+                        var state = entry[nameof(LogHistoryColumn.State)].ToString();
+
+                        if (type == commandType && state == "Committed" &&
+                            DeleteDatabaseCommandGuidIsValid(guid, out var nodes) &&
+                            nodes.Contains(serverStore.NodeTag))
+                        {
+                            if (entry[nameof(LogHistoryColumn.ExceptionType)] != null)
+                            {
+                                if (index == null)
+                                    hasExceptions = true;
+                                continue; // has errors after apply
+                            }
+
+                            hasExceptions = false;
+                            index = curIndex;
+                        }
+                    }
+                }
+            }
+            return index;
+        }
+
+        private static bool DeleteDatabaseCommandGuidIsValid(string guid, out List<string> nodes)
+        {
+            /*the DeleteDatabaseCommand command id should look like that: "db1/A_B_C/a6m8me9843jt"
+                ($"{databaseName}/{string.Join('_', nodes}")/{newGuid}*/
+
+            nodes = null;
+
+            int index1 = guid.IndexOf('/');
+            if (index1 < 0)
+                return false;
+
+            try
+            {
+
+                index1++;
+                var index2 = guid.IndexOf('/', index1);
+                if (index2 < 0)
+                    return false;
+
+                nodes = guid.Substring(index1, index2 - index1).Split('_').Where(x => x != string.Empty).ToList();
+            }
+            catch (ArgumentOutOfRangeException)
```

Comment:
remove this, if we fail unexpectedly here, I want to know why.

Reply:
and safe guard against the case which caused the exception here.

Reply:
Removed